### ISSUE TITLE
Completions docs UI shows docs for wrong completion

### ISSUE THREAD

**Issue by its-dhung** [spyder-ide/spyder#10235]
### What steps reproduce the problem?

1. Start Spyder in debug mode with Kite enabled and running (though I don't think having Kite enabled/running is a requirement)
2. Type the following:
```python
import json
json.dumps$
``` 
3. Highlight the first completion.

### What is the expected output? What do you see instead?

__Expected output:__ I would expect to see the docs for `json.dumps`.

__Actual output:__ I see the docs for `json.dump`.

![image](https://user-images.githubusercontent.com/3752225/65065491-63a99e80-d937-11e9-88d7-52b7f65ead6d.png)

This issue seems intermittent and therefore leads me to believe this is a race condition in Spyder's completions logic.

## Versions

* Spyder version: 4.0.0.dev0
* Python version: 3.7.3 64-bit
* Qt version: 5.9.6
* PyQt version: 5.9.2
* Operating System name/version: Darwin 18.7.0

### Dependencies

```
pygments >=2.0    :  2.4.2 (OK)
qdarkstyle >=2.6.4:  2.7 (OK)
sphinx >=0.6.6    :  2.1.2 (OK)
psutil >=0.3      :  5.6.3 (OK)
pyls >=0.27.0     :  0.28.0 (OK)
nbconvert >=4.0   :  5.5.0 (OK)
pandas >=0.13.1   :  None (NOK)
numpy >=1.7       :  None (NOK)
pympler >=0.7     :  None (NOK)
sympy >=0.7.3     :  None (NOK)
cython >=0.21     :  None (NOK)
qtconsole >=4.5.5 :  4.5.5 (OK)
IPython >=4.0     :  7.8.0 (OK)
matplotlib >=2.0.0:  None (NOK)
pylint >=0.25     :  2.3.1 (OK)
```

**Comment by ccordoba12** [comment]
Thanks @its-dhung for reporting. @goanpeca is going to take a look at it.

**Comment by goanpeca** [comment]
I cannot reproduce this @ccordoba12

**Comment by bcolsen** [comment]
I can reproduce it if I type the last "ps" of `dumps` really fast. It's probably more noticeable on a slow computer.

**Comment by goanpeca** [comment]
Thanks @bcolsen will try to see what we can do about it

**Comment by goanpeca** [comment]
Ok I had to type this ultra fast (beyond what I can really do!) to see it. Hmmm I think we should add a small wait time before we request something to the LSP, the classic 300 ms will do.

This should also solve some of the issues seen with on the fly completion.

**Comment by ccordoba12** [comment]
> the classic 300 ms will do.

Agreed.

**Comment by goanpeca** [comment]
@its-dhung could you test this with the latest master to see if the issue is fixed on your side?

**Comment by ccordoba12** [comment]
@bcolsen, could you help us with this one too?

**Comment by bcolsen** [comment]
It looks fixed to me.

**Comment by ccordoba12** [comment]
Greate, thanks @bcolsen!

**Comment by ccordoba12** [comment]
This was fixed by PR #10262.
